### PR TITLE
Fix the issue with default locale extension for nested classes

### PR DESCRIPTION
**Pull request by simonenkoi**
Fix `@DefaultLocale` for classes annotated with `@Nested`

Lookup of `@DefaultLocale` annotation was added for enclosing class
of the element if it was not found on the element of `ExtensionContext`

Fixes: #146
PR: #161

---

I hereby agree to the terms of the [JUnit Pioneer Contributor License Agreement](https://github.com/junit-pioneer/junit-pioneer/blob/master/CONTRIBUTING.md#junit-pioneer-contributor-license-agreement).

### CODE REVIEW

**Comment by nicolaiparlog** [comment]
Hi Simon, thanks for the PR. I will look at it next Tuesday evening. :)

File: src/main/java/org/junitpioneer/jupiter/Utils.java [hunk 1]
```diff
@@ -12,13 +12,15 @@
...
 import java.util.HashSet;
+import java.util.Optional;
 import java.util.Set;
 import java.util.stream.Collector;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 import org.junit.jupiter.api.extension.ExtensionContext;
 import org.junit.platform.commons.support.AnnotationSupport;
+import org.junit.platform.commons.util.AnnotationUtils;
```
Comment (sormuras) [inline src/main/java/org/junitpioneer/jupiter/Utils.java]: The `org.junit.platform.commons.util` package is not exported by module `org.junit.platform.commons`. It's **not** listed in the API documentation: https://junit.org/junit5/docs/current/api/ and `package-info.java` comments reads:

https://github.com/junit-team/junit5/blob/e80e2854c3862eef3feb08c2dd6276833ef911bd/junit-platform-commons/src/main/java/org/junit/platform/commons/util/package-info.java#L1-L9

Comment (simonenkoi) [reply src/main/java/org/junitpioneer/jupiter/Utils.java]: Good catch, thank you!
I wanted to use the method from AnnotationSupport but somehow one from AnnotationUtils was imported

**Comment by simonenkoi** [comment]
@nicolaiparlog kindly reminder

**Comment by nicolaiparlog** [comment]
I'm starting to work on it [_right now_](https://twitch.tv/nipafx). :smile:

**Comment by nicolaiparlog** [comment]
Thank you very much @simonenkoi, for the fix as well as for the patience. :+1:

**Pull request merged**